readme.md

```diff
@@ -287,6 +287,63 @@ Arguments passed to the CLI will always take precedence over the configuration i
 
 Note that providing files on the CLI overrides the `files` option. If you've configured a glob pattern, for instance `test/**/*.test.js`, you may want to repeat it when using the CLI: `ava 'test/integration/*.test.js'`.
 
+### Using `ava.config.js`
+
+To use an `ava.config.js` file:
+
+ 1. It must be in the same directory as your `package.json`.
+ 2. Your `package.json` must not contain an `ava` property.
+
+The config file can use either ESM or CommonJS syntax. Both of these are valid:
+
+```js
+module.exports = {
+	// …
+};
+```
+
+```js
+export default {
+	// …
+};
+```
+
+The config file may also export a factory function, which will be called with an object containing a `projectDir` property. This function must return a config object, and cannot be a promise.
+
+```js
+const config = ({projectDir}) => {
+	if (projectDir === '...') {
+		return {
+			// Config A
+		};
+	}
+
+	return {
+		// Config B
+	};
+}
+
+export default config;
+```
+
+Or, return different configuration based on the environment: `NODE_ENV=development ava`
```

Comment:
I'm not sure about advocating AVA to be run with different `NODE_ENV` settings. It'd be more interesting to detect CI environments — although at that point we should forward the detection to the function. Perhaps remove for now and track in a follow-up issue?